quiz.py

```diff
@@ -0,0 +1,144 @@
+        return question
+
+
+class ImageQuiz(Quiz):
+    def __init__(self, num_elements_quiz=4):
+        super(ImageQuiz, self).__init__()
+
+        self.elements_dict, self.options_dict = self.initialize_image_quiz(num_elements=10)
+        self.num_elements_quiz = num_elements_quiz
+
+        self.seen_elements_dict = dict()
+        self.current_question = None
+
+    @abstractmethod
+    def initialize_image_quiz(self, num_elements):
+        return
+
+    def generate_question(self):
+        '''
+        Exaustive Algorithm. Shows all images until all are correct.
+        :return: a multiple choice question
+        '''
+        if not self.elements_dict:
+            if not self.seen_elements_dict:
+                return False
+            else:
+                self.elements_dict = self.seen_elements_dict
+                self.seen_elements_dict = dict()
+
+        # 1. grab a key from the element_list randomly
+        correct_key = random.choice(self.elements_dict.keys())
+
+        # 2. Create a subset of the option_list for the quiz
+        correct_element = self.elements_dict[correct_key]
+        correct_option = self.options_dict[correct_key]
+        other_options = filter(lambda x: x != correct_option, set(self.options_dict.values()))
+        aux_list = random.sample(other_options, self.num_elements_quiz - 1)
+        quiz_options_set = set(aux_list)
+        quiz_options_set.add(correct_option)
+
+        # 3. Shuffle the new sub_option_list
+        quiz_options_list = list(quiz_options_set)
+        random.shuffle(quiz_options_list)
+        question = MultipleChoiceQuestion({correct_key: correct_element}, correct_option, quiz_options_list)
+
+        self.current_question = question
+        return question
+
+    def process_answer(self, answer):
+        correct = answer == self.current_question.correct_option.id
+
+        # update score
+        self.update_score(correct)
+
+        # 0. pop element from element list
+        self.elements_dict.pop(self.current_question.question_dict.keys()[0])
+
+        if not correct:
+            # save popped element in the seen_elements list
+            self.seen_elements_dict.update(self.current_question.question_dict)
+
+
+class PainterQuiz(ImageQuiz):
+    def __init__(self, num_elements_quiz=4):
+        super(PainterQuiz, self).__init__(num_elements_quiz=num_elements_quiz)
+
+    def initialize_image_quiz(self, num_elements=5):
+        stmt = exists().where(Painter.id == Painting.painter_id)
+        selected_painters = Painter.query.filter(stmt).order_by(func.random()).limit(num_elements).all()
+
+        painters_dict = dict()
+        paintings_dict = dict()
+        selected_paintings = []
+        for key, painter in enumerate(selected_painters):
+            paintings_aux = Painting.query.filter(Painting.painter_id == painter.id).order_by(func.random()).limit(
+                1).first()
+            selected_paintings.extend([paintings_aux])
```

Comment:
1) `painting_aux` (singular) instead of `paintings_aux`
2) `append` instead of `extend` since you are only adding one element